theano/typed_list/tests/test_basic.py

```diff
@@ -579,5 +579,5 @@ def test_correct_answer(self):
         z = make_list((x, y))
         fc = theano.function([a, b], c)
         fz = theano.function([x, y], z)
-        self.assertTrue(fc(A, B) == [A, B])
-        self.assertTrue(fz(X, Y) == [X, Y])
+        self.assertTrue((m == n).all() for m, n in zip(fc(A, B), [A, B]))
```

Comment:
why this change is needed? It weaken the test. Now we aren't sure both version return an object of the same type. fc could return a list and fz could return a tuple.

I'm not opposed to this change, but I don't understand why it is needed.